geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/filters/HBaseZFilters.scala

```diff
@@ -0,0 +1,67 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.hbase.filters
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.hbase.exceptions.DeserializationException
+import org.apache.hadoop.hbase.filter.{Filter, FilterBase}
+import org.apache.hadoop.hbase.{Cell, CellUtil}
+import org.locationtech.geomesa.index.filters.{Z2Filter, Z3Filter}
+
+class Z3HBaseFilter(filt: Z3Filter) extends FilterBase with LazyLogging {
+
+  override def filterKeyValue(v: Cell): Filter.ReturnCode = {
+    logger.trace("In filterKeyValue()")
+    if (filt.inBounds(v.getRowArray, v.getRowOffset, v.getRowLength)) {
+      Filter.ReturnCode.INCLUDE
+    } else {
+      Filter.ReturnCode.SKIP
+    }
+  }
+
+  override def toByteArray: Array[Byte] = {
+    logger.trace("Serializing Z3HBaseFilter")
+    Z3Filter.toByteArray(filt)
+  }
+}
+
+object Z3HBaseFilter extends LazyLogging {
+  @throws[DeserializationException]
+  def parseFrom(pbBytes: Array[Byte]): Filter = {
```

Comment:
seems like there is no java equivalent so it must be finding this - @anthonyccri can you confirm?